src/diffusers/utils/peft_utils.py

```diff
@@ -147,11 +147,17 @@ def unscale_lora_layers(model, weight: Optional[float] = None):
                     module.set_scale(adapter_name, 1.0)
 
 
-def get_peft_kwargs(rank_dict, network_alpha_dict, peft_state_dict, is_unet=True):
+def get_peft_kwargs(rank_dict, network_alpha_dict, peft_state_dict, config=None, is_unet=True):
     rank_pattern = {}
     alpha_pattern = {}
     r = lora_alpha = list(rank_dict.values())[0]
 
+    # Try to retrive config.
+    alpha_retrieved = False
+    if config is not None:
+        lora_alpha = config["lora_alpha"]
```

Comment:
I think this also needs to consider [`alpha_pattern`](https://huggingface.co/docs/peft/v0.12.0/en/package_reference/lora#peft.LoraConfig.alpha_pattern) -- if not intended to support for now, at least raise an error if `alpha_pattern` is given? If support is added, the unit test should include different alpha values (or a separate test could be added for this).

Reply:
`alpha_pattern` cannot be provided through `get_peft_kwargs`.

Reply:
Hmm, maybe I'm misunderstanding, but we're passing a `LoraConfig` instance, how can we know that this does not have `alpha_pattern`?

I wonder if this should be:

```diff
    # Try to retrieve config.
    alpha_retrieved = False
    if config is not None:
        lora_alpha = config["lora_alpha"] if "lora_alpha" in config else lora_alpha
        alpha_retrieved = True
+       if config.get("alpha_pattern", None):
+           alpha_pattern = config["alpha_pattern"]
```

Similar argument for `rank_pattern`. In general, it's not clear to me how we should handle it if `rank_pattern`/`alpha_pattern` differ from `rank_dict`/ `network_alpha_dict` (or is it not possible)?

Reply:
Good point. Let me accommodate those changes and have it ready for your review.

Reply:
@BenjaminBossan does https://github.com/huggingface/diffusers/pull/9143/commits/f7d30de31a95e0eda3bd2da250fe924b08c4aa3c work?

Reply:
Looks good, just wondering about the warning:

> "`alpha_pattern` found in the LoRA config. This will be ignored."

Do you mean:  "\`network_alpha_dict\` will be ignored"? Same question for rank pattern.

Reply:
No. The `alpha_pattern` provided in the config will be ignored because we are calculating it just a few line below.

Reply:
Okay, I was confused because you were assigning it, which was unnecessary if it was ignored. But still, `alpha_pattern` is only assigned below `if not alpha_retrieved`, but here we set `alpha_retrieved = True`, so right now if `config` is passed, `alpha_pattern` is neither set by the config, nor via `network_alpha_dict`.

Reply:
I have removed the assignment. Does that solve your concern?

Reply:
No, IIUC there is still an issue. Say `get_peft_kwargs` is called with both `network_alpha_dict` and `config` being passed. As is, `network_alpha_dict` is completely ignored and `alpha_pattern` remains an empty dict. I think what should happen is that either

1. `alpha_pattern` is taken from `network_alpha_dict` (same as previously)
2. `alpha_pattern` is taken from `config` if `config.alpha_pattern` is not `None`. If it is `None`, `network_alpha_dict` should be used.

So either `network_alpha_dict` or `config.alpha_pattern` should take precedence. And if both are given, potentially warn about the one being ignored. WDYT?

Depending on the choice here, `rank_pattern` should also be adjusted for consistency.

Reply:
> alpha_pattern is taken from config if config.alpha_pattern is not None. If it is None, network_alpha_dict should be used.

This is not what I am doing. That is because `network_alpha_dict` can only be true for non-diffusers checkpoints. For those checkpoints, metadata won't have a PEFT config.  

Long story cut, short, `rank_pattern` and `alpha_pattern` from config (if found) will be simply ignored for now.

Reply:
Thanks for clarifying, so it is assumed that `rank_pattern`/`alpha_pattern` are never passed when `config` is passed, and vice versa. In that case, this could be checked and an error raised, or at least a comment added, WDYT?

Reply:
Will add a comment. For now, a warning like we are doing now would suffice.

Reply:
Added a comment in 178a4596b.